src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.cs

```diff
@@ -165,11 +166,13 @@ private static SymmetricAlgorithm OpenAlgorithm(string algorithmIdentifier)
 
             switch (algorithmIdentifier)
             {
-                case Oids.Rc2Cbc:
+                case Oids.Rc2Cbc when Helpers.IsRC2Supported:
```

Comment:
Nit - this feels weird to me to have `case x when y:` followed by `case x:` as a fallback. Having a standard construct which doesn't use the _when_ clause would help readability IMO. But if others disagree with this statement then I'll relent. :)

```cs
case Oids.Rc2Cbc:
if (!supported) { /* throw */ }
alg = RC2.Create();
break;
```

Reply:
Oh, good, it's not just me.

I know that I don't like a lot of the new language constructions, so complaining about their lack of legibility makes me feel like a luddite.  But I'm happy to +1 someone else :smile:

Reply:
Eh, I'm happy to avoid nesting conditional logic as much as possible (`if` in a `case` always looks weird to me) but I'm just a guest in your house 😄. Fixed.